packages/flutter_tools/lib/src/base/utils.dart

```diff
@@ -152,9 +152,7 @@ class SettingsFile {
     }
   }
 
-  factory SettingsFile.parseFromFile(File file) {
-    return SettingsFile.parse(file.readAsStringSync());
-  }
+  SettingsFile.parseFromFile(File file) : this.parse(file.readAsStringSync());
```

Comment:
while it seems like this still does the right thing, I would say this code is very confusing. We are invoking a named constructor strictly for the side effects and discarding the return value?

Can you either revert this diff, or inline the body of `SettingsFile.parse` to this constructor body?

Reply:
Just some context: This is a redirecting constructor, a (not very well known and hence underutilized) feature in dart:
* https://dart.dev/language/constructors#redirecting-constructors
* https://stackoverflow.com/questions/20015249/in-dart-can-you-call-another-constructor-from-a-constructor

The return value of the `this.parse` constructor is not discarded, it is the return value of the `SettingsFile.parseFromFile` constructor.

Of course, it's up to us to decide whether we want to use this feature in our code base or if we think it makes things harder to read. Personally, I am OK with it since it is not that much different from calling a super constructor:

```dart
Foo(int b) : super.namedConstructor(a: b);
```

Reply:
ahh, ok, I read about this at some point and promptly forgot. This is fine.